File: growing-shop-auth-before/src/main/java/com/example/growingshopauth/company/service/CompanyService.java
package com.example.growingshopauth.company.service;

import com.example.growingshopauth.company.domain.Company;
import com.example.growingshopauth.company.dto.CompanyResponse;
import com.example.growingshopauth.company.repository.CompanyRepo;
import jakarta.persistence.EntityNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CompanyService {
    private final CompanyRepo companyRepo;

    public CompanyResponse.CompanyRes getCompany(Long id) {
        Company company = companyRepo.findById(id)
                .orElseThrow(() -> new EntityNotFoundException(id + " 에 해당하는 업체가 존재하지 않습니다."));

        return CompanyResponse.CompanyRes.from(company);
    }
}

File: growing-shop-api/auth/src/main/kotlin/com/example/growingshopauth/company/service/CompanyService.kt
package com.example.growingshopauth.company.service

import com.example.growingshopauth.company.dto.CompanyResponse
import com.example.repository.company.CompanyRepository
import jakarta.persistence.EntityNotFoundException
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class CompanyService(
    private val companyRepository: CompanyRepository
) {

    fun getCompany(id: Long): CompanyResponse.CompanyRes = CompanyResponse.CompanyRes.from(
        companyRepository.findById(id)
            .orElseThrow { EntityNotFoundException("$id 에 해당하는 업체가 존재하지 않습니다.") }
    )
}
